shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/modules/login/LoginViewModel.kt

```diff
@@ -0,0 +1,41 @@
+package co.nimblehq.blisskmmic.presentation.modules.login
+
+import co.nimblehq.blisskmmic.data.network.helpers.toErrorMessage
+import co.nimblehq.blisskmmic.domain.usecase.LogInUseCase
+import co.nimblehq.blisskmmic.presentation.modules.ViewModel
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+data class LoginViewState(
```

Comment:
@blyscuit Can we define global `ViewState` and apply for Login?
For example:

```
sealed class ViewState {
    object Success : ViewState()
    object Fail : ViewState()
    data class Error(val message: String?) : ViewState()
}
........
 private val mutableViewState: MutableStateFlow<ViewState ?> =
        MutableStateFlow()

    val viewState: StateFlow<ViewState ?> = mutableViewState
```

Reply:
I did not do this way because each ViewState will have different properties and not all will have Success, Fail, Error, as well as having extra states that do not match each other. Hence `private val mutableViewState: MutableStateFlow<CurrrentViewState>` will always be different.